addons/calendar_bis/models/calendar_timeslot.py

```diff
@@ -0,0 +1,234 @@
+        elif self.freq in ['daily', 'yearly']:
+            self.event_id.make_timeslots()
+        elif self.freq == 'weekly':
+            self.event_id.write({
+                days[(dt - delta).weekday()]: False,
+                days[dt.weekday()]: True,
+            })
+            # write will trigger make_timeslots
+        elif self.freq == 'monthly':
+            self.monthday = dt.day
+            # TODO monthweekday
+            self.event_id.make_timeslots()
+
+    # RECURRING RELATED
+    @api.depends('event_id')
+    def _compute_recurring(self):
+        for timeslot in self:
+            timeslot.update({
+                'is_recurring': timeslot.event_id.is_recurring,
+                'mo': timeslot.event_id.mo,
+                'tu': timeslot.event_id.tu,
+                'we': timeslot.event_id.we,
+                'th': timeslot.event_id.th,
+                'fr': timeslot.event_id.fr,
+                'sa': timeslot.event_id.sa,
+                'su': timeslot.event_id.su,
+                'freq': timeslot.event_id.freq,
+                'until': timeslot.event_id.until,
+                'count': timeslot.event_id.count,
+                'interval': timeslot.event_id.interval,
+                'monthday': timeslot.event_id.monthday,
+                'monthweekday_n': timeslot.event_id.monthweekday_n,
+                'monthweekday_day': timeslot.event_id.monthweekday_day,
+            })
+
+    def _inverse_recurring(self):
+        for timeslot in self:
+            timeslot.event_id.write({
+                'is_recurring': timeslot.is_recurring,
+                'mo': timeslot.mo,
+                'tu': timeslot.tu,
+                'we': timeslot.we,
+                'th': timeslot.th,
+                'fr': timeslot.fr,
+                'sa': timeslot.sa,
+                'su': timeslot.su,
+                'freq': timeslot.freq,
+                'until': timeslot.until,
+                'count': timeslot.count,
+                'interval': timeslot.interval,
+                'monthday': timeslot.monthday,
+                'monthweekday_n': timeslot.monthweekday_n,
+                'monthweekday_day': timeslot.monthweekday_day,
+            })
+
+    # COMPUTES
+    @api.depends('duration')
+    def _compute_stop(self):
+        for slot in self:
+            slot.stop = slot.start + relativedelta(hours=slot.duration)
+
+    @api.depends('stop', 'start')
+    def _compute_duration(self):
+        for slot in self:
+            slot.duration = (slot.stop - slot.start).total_seconds() / 3600
+
+    def copy(self, values):
+        if 'attendee_ids' not in values:
+            values['attendee_ids'] = [Command.create(attendee.copy_data()[0]) for attendee in self.attendee_ids]
+        return super().copy(values)
+
+    def copy_data(self, default=None):
+        data = super().copy_data(default)[0]
+        return [{**data, 'attendee_ids': [Command.create(attendee.copy_data()[0]) for attendee in self.attendee_ids]}]
```

Comment:
copy=True on attendee_ids should do the trick + copy=False on attendee fields you don't want to copy (state, ...)